### ISSUE TITLE
Print information message to console if Dummy Audio driver is loaded.

### ISSUE THREAD

**Issue by merriam** [godotengine/godot#42630]
**Godot version:**
Godot Engine v3.2.2.stable.official - https://godotengine.org

**OS/device including version:**
any

**Issue description:**
If the dummy audio driver is loaded, the console should print that fact.  Otherwise, it is exceptionally hard to diagnose why a different audio driver is loaded.   The dummy audio driver was added in #20986 (but not yet in docs).

**Steps to reproduce:**
Take a project, set Project/Project Settings/Application/Audio/Driver to `Foobar` and restart.
Run any project with audio.
Did the dummy audio driver load?

<img width="614" alt="Screen Shot 2020-10-07 at 11 46 40 PM" src="https://user-images.githubusercontent.com/159998/95424711-e82ade80-08f7-11eb-9f44-f5a5ddd4b011.png">

**Comment by merriam** [comment]
Hmmm..... 

It looks like the fix would be to change servers/audio_server.cpp around line 213 from:

```
	if (driver_count > 1 && String(AudioDriver::get_singleton()->get_name()) == "Dummy") {
		WARN_PRINT("All audio drivers failed, falling back to the dummy driver.");
	}
```

to

```
	if (driver_count > 1 && String(AudioDriver::get_singleton()->get_name()) == "Dummy") {
		WARN_PRINT("All audio drivers failed, falling back to the dummy driver.");
	} else if (String(AudioDriver::get_singleton()->get_name()) == "Dummy")) {
                WARN_PRINT("Using the dummy audio driver.");
       }
```

It would be awsome if someone with the dev environment could test this and put in a PR?

**Comment by merriam** [comment]
Gah! Somewhat of a tangent, but is the Project Settings/Application/Audio/Driver setting even used?  It looks like the code just ignores it and takes the core audio driver anyway.   Is there any documentation as to what is supposed to happen with this setting?

**Comment by Calinou** [comment]
> Gah! Somewhat of a tangent, but is the Project Settings/Application/Audio/Driver setting even used? It looks like the code just ignores it and takes the core audio driver anyway. Is there any documentation as to what is supposed to happen with this setting?

It's used, but it might not work on all platforms: https://github.com/Orama-Interactive/Pixelorama/pull/312

**Comment by bruvzg** [comment]
> Take a project, set Project/Project Settings/Application/Audio/Driver to Foobar and restart.
Run any project with audio.
Did the dummy audio driver load?

Setting invalid driver name should not cause fallback to "Dummy", but use default one. "Dummy" is used only if it is selected in the settings or when all other drivers failed to load.

**Comment by merriam** [comment]
TL;DR:  We should print the "using dummy audio driver" warning but also need better documentation.

1. The bug report is incorrect, in that changing the driver to 'Foo' is same as leaving the driver blank.  Changing the driver name to "dummy" does load the dummy audio driver.  The exact message could be improved:  "No audio (using dummy audio driver)"

2.  The documentation in project settings incorrect or incomplete.  One view is something like:
     "Specifies the audio driver to use.  The value is platform specific, though a blank or unknown value uses the platform default.  The value 'dummy' uses a the dummy audio driver  which reduces CPU load by not producing sound.  Requires restarting the editor for changes to take effect."

      If the documentation is wrong, please let me know.  I'll try to link to a better PR if its just the documentation.

**Comment by Calinou** [comment]
Should this message only be printed in the editor Output panel, or also in projects? This may be annoying in projects that rely on the Dummy audio driver but are not headless (as that would result in additional non-removable CLI boilerplate output).
